### PR TITLE
【hydra No.15】deeponet

### PR DESCRIPTION
**Pull request by co63oc**
### PR types

Others 
### PR changes

Others 
### Describe

https://github.com/PaddlePaddle/PaddleScience/issues/571

15 deeponet

因数据集需要注册yale.edu后缀邮箱下载，无法获取数据集，只是修改代码没有运行测试
image

### CODE REVIEW

File: docs/zh/examples/deeponet.md [hunk 1]
```diff
@@ -2,6 +2,26 @@
 
 <a href="https://aistudio.baidu.com/aistudio/projectdetail/6566389?sUid=438690&shared=1&ts=1690775701017" class="md-button md-button--primary" style>AI Studio快速体验</a>
 
+=== "模型训练命令"
+
+    ``` sh
+    # linux
+    wget https://paddle-org.bj.bcebos.com/paddlescience/datasets/deeponet/antiderivative_unaligned_train.npz
```
Comment (HydrogenSulfate) [inline docs/zh/examples/deeponet.md:9]: 需要下载以下两个文件：
https://paddle-org.bj.bcebos.com/paddlescience/datasets/DeepONet/antiderivative_unaligned_test.npz
https://paddle-org.bj.bcebos.com/paddlescience/datasets/DeepONet/antiderivative_unaligned_train.npz

File: docs/zh/examples/deeponet.md [hunk 2]
```diff
@@ -2,6 +2,26 @@
...
+    ```
+
+=== "模型评估命令"
+
+    ``` sh
+    # linux
+    wget https://paddle-org.bj.bcebos.com/paddlescience/datasets/deeponet/antiderivative_unaligned_train.npz
+    # windows
+    # curl https://paddle-org.bj.bcebos.com/paddlescience/datasets/deeponet/antiderivative_unaligned_train.npz --output antiderivative_unaligned_train.npz
+    python deeponet.py mode=eval EVAL.pretrained_model_path=https://paddle-org.bj.bcebos.com/paddlescience/models/deeponet/deeponet_pretrained.pdparams
```
Comment (HydrogenSulfate) [inline docs/zh/examples/deeponet.md:22]: 预训练模型地址为：https://paddle-org.bj.bcebos.com/paddlescience/models/deeponet/deeponet_pretrained.pdparams

File: examples/operator_learning/conf/deeponet.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  trunk_hidden_size: 40
+  branch_activation: relu
+  trunk_activation: relu
+  use_bias: true
+
+# training settings
+TRAIN:
+  epochs: 10000
+  learning_rate: 1.0e-3
+  solver:
```
Comment (HydrogenSulfate) [inline examples/operator_learning/conf/deeponet.yaml:51]: solver: 字段可以删除，yaml的字段虽然填写限制不多，但尽量少一些嵌套

File: examples/operator_learning/conf/deeponet.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    epochs: ${TRAIN.epochs}
+    iters_per_epoch: 1
+    save_freq: 500
+    eval_freq: 500
+    log_freq: 20
+    eval_during_train: true
+    seed: 2023
```
Comment (HydrogenSulfate) [inline examples/operator_learning/conf/deeponet.yaml:58]: 重复的seed字段删除

File: examples/operator_learning/conf/deeponet.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  branch_activation: relu
+  trunk_activation: relu
+  use_bias: true
+
+# training settings
+TRAIN:
+  epochs: 10000
+  learning_rate: 1.0e-3
+  solver:
+    epochs: ${TRAIN.epochs}
```
Comment (HydrogenSulfate) [inline examples/operator_learning/conf/deeponet.yaml:52]: 重复的epochs字段删除

File: examples/operator_learning/conf/deeponet.yaml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    iters_per_epoch: 1
+    save_freq: 500
+    eval_freq: 500
+    log_freq: 20
```
Comment (HydrogenSulfate) [inline examples/operator_learning/conf/deeponet.yaml:56]: log_req移动至output_dir字段下方，与bracket.yaml平级，因为这个字段trian和eval共用

File: examples/operator_learning/conf/deeponet.yaml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    eval_during_train: true
+    seed: 2023
+    eval_with_no_grad: true
```
Comment (HydrogenSulfate) [inline examples/operator_learning/conf/deeponet.yaml:59]: TRAIN.eval_with_no_grad是多余字段，删除

File: examples/operator_learning/deeponet.py [hunk 8]
```diff
@@ -88,42 +68,121 @@
...
-        EPOCHS,
-        ITERS_PER_EPOCH,
-        save_freq=EVAL_FREQ,
-        log_freq=20,
-        eval_during_train=True,
-        eval_freq=EVAL_FREQ,
-        seed=SEED,
         validator=validator,
-        eval_with_no_grad=True,
+        **cfg.TRAIN.solver,
```
Comment (HydrogenSulfate) [inline examples/operator_learning/deeponet.py:78]: 按照修改前的字段逐个填写，这里不用**，减少代码阅读难度

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: examples/operator_learning/deeponet.py (hunk 8 above)
Comment (co63oc) [reply examples/operator_learning/deeponet.py]: 已修改solver

File: docs/zh/examples/deeponet.md [hunk 9]
```diff
@@ -113,39 +137,39 @@ examples/operator_learning/deeponet.py:57:61
...
-接下来我们需要指定训练轮数和学习率，此处我们按实验经验，使用十万轮训练轮数，并每隔 500 个 epochs 评估一次模型精度。
+接下来我们需要指定训练轮数和学习率，此处我们按实验经验，使用一万轮训练轮数，并每隔 500 个 epochs 评估一次模型精度。
 
-``` py linenums="65"
+``` py linenums="49"
 --8<--
-examples/operator_learning/deeponet.py:65:66
+examples/operator_learning/conf/deeponet.yaml:49:50
 --8<--
 ```
```
Comment (HydrogenSulfate) [inline docs/zh/examples/deeponet.md:154]:
``` suggestion
``` yaml linenums="49"
--8<--
examples/operator_learning/conf/deeponet.yaml:49:55
--8<--
```
```

**Review by HydrogenSulfate — CHANGES_REQUESTED** [review]

File: docs/zh/examples/deeponet.md (hunk 9 above)
Comment (co63oc) [reply docs/zh/examples/deeponet.md]: 已修改

**Review by HydrogenSulfate — APPROVED** [review]
LGTM
